src/datamodel/builder/Uuid.cpp

```diff
@@ -10,3 +10,99 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Uuid.h"
+
+#include <random>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::builder {
+
+struct UuidBuilder::UuidSharedState {
+	std::mt19937_64 random_engine{std::random_device{}()};
```

Comment:
Is this thread safe? Maybe the best thing to do is add a constructor to this struct that generates rand_b then this member and the `rand_b_initialized` member can be removed.

Making that change along with the `std::call_once` in another comment would make the initialization of rand_b safe.

Reply:
We can hold off on making the shared state thread safe for this PR